elisa/src/client/RunDetails/RunDetails.tsx

```diff
@@ -0,0 +1,245 @@
+    SAMPLE_COL_NAME, SAMPLE_COLUMN_NAMES, SPOT_COL_NAME
+} from "./constants";
+import { CurveFitData, PlotOptions } from "./models";
+import {
+    filterDataByPlotOptions,
+    getColumnInfoFromQueryDetails,
+    getDefaultPlotOptions,
+    getMissingFields,
+    getUniqueIdsForPlotSelections,
+    getUniqueValues,
+    getUpdatedPlotOptions,
+    parsePlotDataFromResponse,
+    shouldReloadCurveFitData
+} from "./utils";
+import { getCurveFitXYPairs } from "./actions";
+import { CalibrationCurvePanel } from "./components/CalibrationCurvePanel";
+import { DataSelectionsPanel } from "./components/DataSelectionsPanel";
+import { PlotOptionsPanel } from "./components/PlotOptionsPanel";
+import { CurveFitPanel } from "./components/CurveFitPanel";
+
+import './RunDetails.scss';
+
+export interface AppContext {
+    protocolId: number,
+    runId: number,
+    schemaName: string,
+}
+
+interface Props {
+    context: AppContext
+}
+
+interface State {
+    runPropertiesRow: {[key: string]: any},
+    runPropertiesError: string
+    data: any[],
+    dataError: string,
+    columnInfo: {[key: string]: any},
+    measures: string[]
+}
+
+export class RunDetails extends PureComponent<Props, State> {
+    state: Readonly<State> = {
+        runPropertiesRow: undefined,
+        runPropertiesError: undefined,
+        data: undefined,
+        dataError:  undefined,
+        columnInfo: {},
+        measures: [DEFAULT_X_AXIS_PROP, DEFAULT_Y_AXIS_PROP]
+    };
+
+    componentDidMount(): void {
+        const { schemaName, runId } = this.props.context;
+
+        Query.getQueryDetails({
+            schemaName,
+            queryName: 'Data',
+            success: (queryInfo) => {
+                const columnInfo = getColumnInfoFromQueryDetails(queryInfo);
+                this.setState(() => ({
+                    columnInfo: columnInfo.columnInfo,
+                    measures: columnInfo.measures
+                }), () => {
+                    this.queryData();
+                });
+            },
+            failure: (reason) => {
+                console.error(reason);
+                this.setState(() => ({ dataError: reason.exception }));
+            }
+        });
+
+        Query.selectRows({
+            schemaName,
+            queryName: 'Runs',
+            columns: RUN_COLUMN_NAMES.join(','),
+            filterArray: [Filter.create('RowId', runId)],
+            success: (data) => {
+                if (data?.rows?.length === 1) {
+                    this.setState(() => ({ runPropertiesRow: data.rows[0] }));
+                }
+                else {
+                    this.setState(() => ({ runPropertiesError: 'No properties found for run ID ' + runId + '.' }));
+                }
+            },
+            failure: (reason) => {
+                console.error(reason);
+                this.setState(() => ({ runPropertiesError: reason.exception }));
+            }
+        });
+    }
+
+    queryData() {
```

Comment:
This is called in a callback in a callback, seems like it should be arrow function to ensure scope.  And missing return type.

Reply:
Done